pkg/host-disk/host-disk_test.go

```diff
@@ -368,90 +368,116 @@ var _ = Describe("HostDisk", func() {
-			}
-
+			Expect(vmi.Spec.Volumes).To(HaveLen(1), "There should always be 1 volume")
+			validateFunc()
 		},
 
-			Entry("filemode", k8sv1.PersistentVolumeFilesystem, "disk"),
-			Entry("blockmode", k8sv1.PersistentVolumeBlock, "disk"),
-			Entry("filesystem passthrough", k8sv1.PersistentVolumeFilesystem, "filesystem"),
+			Entry("filemode", k8sv1.PersistentVolumeFilesystem,
+				v1.Devices{},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				func() {
+					ExpectedCapacity := resource.MustParse("2Gi")
+					Expect(vmi.Spec.Volumes[0].HostDisk).NotTo(BeNil(), "There should be a hostdisk volume")
+					Expect(vmi.Spec.Volumes[0].HostDisk.Type).To(Equal(v1.HostDiskExistsOrCreate), "Correct hostdisk type")
+					Expect(vmi.Spec.Volumes[0].HostDisk.Path).NotTo(BeNil(), "Hostdisk path is filled")
+					Expect(vmi.Spec.Volumes[0].HostDisk.Capacity.Value()).To(Equal(ExpectedCapacity.Value()), "Hostdisk capacity is filled")
+					Expect(vmi.Spec.Volumes[0].PersistentVolumeClaim).To(BeNil(), "There shouldn't be a PVC volume anymore")
+				},
+			),
+			Entry("filemode with smaller requested PVC size", k8sv1.PersistentVolumeFilesystem,
+				v1.Devices{},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("1Gi")},
+				func() {
+					ExpectedCapacity := resource.MustParse("1Gi")
+					Expect(vmi.Spec.Volumes[0].HostDisk).NotTo(BeNil(), "There should be a hostdisk volume")
+					Expect(vmi.Spec.Volumes[0].HostDisk.Type).To(Equal(v1.HostDiskExistsOrCreate), "Correct hostdisk type")
+					Expect(vmi.Spec.Volumes[0].HostDisk.Path).NotTo(BeNil(), "Hostdisk path is filled")
+					Expect(vmi.Spec.Volumes[0].HostDisk.Capacity.Value()).To(Equal(ExpectedCapacity.Value()), "Hostdisk capacity is filled")
+					Expect(vmi.Spec.Volumes[0].PersistentVolumeClaim).To(BeNil(), "There shouldn't be a PVC volume anymore")
+				},
+			),
+			Entry("blockmode", k8sv1.PersistentVolumeBlock,
+				v1.Devices{},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				func() {
+					Expect(vmi.Spec.Volumes[0].HostDisk).To(BeNil(), "There should be no hostdisk volume")
+					Expect(vmi.Spec.Volumes[0].PersistentVolumeClaim).ToNot(BeNil(), "There should still be a PVC volume")
+					Expect(vmi.Spec.Volumes[0].PersistentVolumeClaim.ClaimName).To(Equal(pvcName), "There should still be the correct PVC volume")
+				},
+			),
+			Entry("filesystem passthrough", k8sv1.PersistentVolumeFilesystem,
+				v1.Devices{
+					Filesystems: []v1.Filesystem{{
+						Name:     volumeName,
+						Virtiofs: &v1.FilesystemVirtiofs{},
+					}},
+				},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				k8sv1.ResourceList{k8sv1.ResourceStorage: resource.MustParse("2Gi")},
+				func() {
```

Comment:
wouldn't it be better to define these validation functions earlier in the test - readability wise?